### PR TITLE
[neuralnet] check tflite interpreter is enabled when export

### PR DESCRIPTION
**Pull request by lhs8928**
- Added ifdef statement to check tflite interpreter is enabled
 - Added override specifier in get/setWeights

Signed-off-by: hyeonseok lee <hs89.lee@samsung.com>

### CODE REVIEW

File: nntrainer/models/neuralnet.cpp [hunk 1]
```diff
@@ -1203,16 +1203,20 @@ void NeuralNetwork::exports(const ml::train::ExportMethods &method,
...
+
+      /// We will call "serialize" method for the model which is already trained
+      /// or allocated. So, we need to call deallocateTensors first to make sure
+      /// `dealloc_weights == false`
+      model_graph.deallocateTensors();
+      model_graph.allocateTensors(ExecutionMode::INFERENCE);
+      interpreter.serialize(graph_representation, file_path);
+      model_graph.deallocateTensors();
+    #else
+      throw std::runtime_error{"ENABLE_TFLITE_INTERPRETER is not defined"};
```
Comment (songgot) [inline nntrainer/models/neuralnet.cpp:1217]: the log indicates how to correct the error.
I think that app set 'METHOD_TFLITE' and  it is defined in enum class (common.h) as below
enum class ExportMethods {
  METHOD_STRINGVECTOR = 0, /**< export to a string vector */
  METHOD_TFLITE = 1,       /**< epxort to tflite */
  METHOD_UNDEFINED = 999,  /**< undefined */
};
If we focus on 'METHOD_TFLITE', how about change error log?
```suggestion
throw std::runtime_error{"METHOD_TFLITE is not supported"};
throw std::runtime_error{"METHOD_TFLITE is not supported , please define ENABLE_TFLITE_INTERPRETER"};
```

**Review by DonghakPark — APPROVED** [review]
LGTM!

File: nntrainer/models/neuralnet.cpp (hunk 1 above)
Comment (lhs8928) [reply nntrainer/models/neuralnet.cpp]: Done :)

**Review by songgot — APPROVED** [review]
LGTM

**Review by jijoongmoon — APPROVED** [review]
LGTM

**Pull request merged**